src/nextjournal/clerk.clj

```diff
@@ -58,53 +58,83 @@
      {:result ret#
       :time-ms (/ (double (- (. System (nanoTime)) start#)) 1000000.0)}))
 
+(defn- var-from-def [var]
+  (let [resolved-var (cond (var? var)
+                           var
+
+                           (symbol? var)
+                           (find-var var)
+
+                           :else
+                           (throw (ex-info "Unable to resolve into a variable" {:data var})))]
+    {::var-from-def resolved-var}))
+
+(defn- lookup-cached-result [results-last-run introduced-var hash cas-hash visibility]
+  (try
+    (let [value (or (get results-last-run hash)
+                    (thaw-from-cas cas-hash))]
+      (when (and introduced-var (not (::var-from-def value)))
+        (intern *ns* (-> introduced-var symbol name symbol) value))
```

Comment:
the `results-last-run` for `(def x 1)` will point to `{::var-from-def #'x}` and then without the `(not (::var-from-def value))` we would `(intern *ns* #'x {::var-from-def #'x})` which causes the viewer code to stackoverflow when it it tries to display this var.

I'm unsure what the best approach to caching vars that we want to leave unresolved until viewer time (to allow for test running for example). So I just added a skip here. I think that the cache-to-filesystem code still caches the var->value even if it comes from a def, we just never use it.